### PR TITLE
Remove Tslint disable comment for BaseStore interface

### PR DESCRIPTION
**Pull request by Shobhit1**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`npm run test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

#### Description of changes

Remove Tslint disable comment for BaseStore interface

### CODE REVIEW

File: src/background/stores/base-store.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { Store } from '../../common/flux/store';
-import { IBaseStore } from '../../common/istore';
+import { BaseStore } from '../../common/istore';
 import { StoreNames } from '../../common/stores/store-names';
 
-export abstract class BaseStore<TState> extends Store implements IBaseStore<TState> {
+export abstract class BaseStoreImpl<TState> extends Store implements BaseStore<TState> {
```
Comment (smoralesd) [inline src/background/stores/base-store.ts]: I'm wondering if we should update the file name too... it should only affect the files that are already part of this PR, right?

Comment (Shobhit1) [reply src/background/stores/base-store.ts]: I think so. I will check and update this PR.

File: src/common/istore.d.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-// tslint:disable-next-line:interface-name
-export interface IBaseStore<TState> {
+
+export interface BaseStore<TState> {
```
Comment (smoralesd) [inline src/common/istore.d.ts]: should we change the file name here? and move from `.d.ts` to `.ts` is a good idea too

Comment (Shobhit1) [reply src/common/istore.d.ts]: I want to, but  again only if It doesn't increase the number of changes too much!

Comment (Shobhit1) [reply src/common/istore.d.ts]: Done. Changed it to `base-store.ts`

File: src/background/stores/base-store.ts (hunk 1 above)
Comment (Shobhit1) [reply src/background/stores/base-store.ts]: changed it to `base-store-impl.ts`

**Comment by Shobhit1** [comment]
> I have some comments about file names

Done. Thanks for the feedback. 😄

**Pull request merged**